sdk/provisioning/Azure.Provisioning/src/Construct.cs

```diff
@@ -103,7 +110,7 @@ public IEnumerable<Resource> GetResources(bool recursive = true)
             IEnumerable<Resource> result = _resources;
             if (recursive)
             {
-                result = result.Concat(GetConstructs(false).SelectMany(c => c.GetResources(false)));
+                result = result.Concat(GetConstructs(true).SelectMany(c => c.GetResources(false)));
```

Comment:
I am really curious why this is needed here but not for parameters and outputs?

Reply:
I think it is needed for Parameters but not GetConstructs as that already has recursion built in.

Reply:
It is needed for the others, but for those we will need to dedup the results. For resources, I'm not sure that dedup is needed as a resource shouldn't belong to multiple constructs.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/42044